### PR TITLE
WIP: filterbar documentation

### PR DESCRIPTION
**Pull request by MFSepplive**
First Version of the filterbar documentation, mainly to get some feedback about it :)

### CODE REVIEW

**Comment by nsams** [comment]
FilterBar is a feature almost exclusively used above a Table to act as it's filter. I thinks it's docs should also be in that direction.

It might be true that you can use it in any final-form, but even it's name suggests nobody will ever do that.

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+
+<Meta title="Docs/Components/Table/Filterbar/FilterBar" />
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:3]: Filterbar or FilterBar? We should be consistent here. I'm rooting for FilterBar

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+const ColorFilterField: React.FC<ColorFilterFieldProps> = ({ colors }) => {
+    const options = colors
+        .filter((color, index, colorsArray) => colorsArray.indexOf(color) == index) //filter colorsArray to only have unique values as select options
+        .map((color) => {
+            return { value: color, label: color };
+        });
+
+    return <Field name="color" type="text" component={FinalFormReactSelectStaticOptions} fullWidth options={options} />;
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:31]: We're trying to get rid of react-select. Therefore we shouldn't use it in new stories, in my opinion

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+onSubmit={(values) => {
+    for (const name in values) {
+        outerForm.change(name, values[name]);
+    }
+}}
+```
+
+The FilterBarActiveFilterBadge shows the current count of set values of every FilterBarPopoverFilter form. This componentas well as the calculation can be overritten.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:38]:
```suggestion
The FilterBarActiveFilterBadge shows the current count of set values of every FilterBarPopoverFilter form. This component as well as the calculation can be overridden.
```

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Table Example
+
+<Canvas>
+    <Story id="stories-components-table-filterbar-filterbar-default--default" />
+</Canvas>
+
+## Advanced Informations
+
+Every FilterBarPopoverFilter Component contains its own FinalForm. So the Filterbar is kinda like a collection of final forms, which pass their values into a wrapping form.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:26]: This is an implementation detail. I don't think potential users need to know this

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+
+<Meta title="Docs/Components/Table/Filterbar/FilterBar" />
+
+# Filterbar
+
+The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include all kind of sorts of Final Form Fields.
+The Intention of the Filterbar was to act as a Filter of a Table, which is also the main usecase for this Component.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:8]:
```suggestion
The intended use case for the FilterBar is to act as a filter on top of a table.
```

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+
+<Meta title="Docs/Components/Table/Filterbar/FilterBar" />
+
+# Filterbar
+
+The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include all kind of sorts of Final Form Fields.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:7]: "all kind of sorts" is to casual for documentation

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        horsepower: faker.datatype.number({ min: 50, max: 200 }),
+        price: faker.commerce.price(100, 1000, 2),
+        owner: {
+            firstname: faker.name.firstName(),
+            lastname: faker.name.lastName(),
+        },
+    };
+});
+
+storiesOf("stories/components/Table/Filterbar/Filterbar/Default", module).add("Default", () => {
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:81]: In my opinion, 120 LOC for the default story of a component is too long

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx (hunk 1 above)
Comment (MFSepplive) [reply packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:3]: yes thats true :) i am also tending to FilterBar.

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx (hunk 2 above)
Comment (MFSepplive) [reply packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:31]: i will change it to the new Select :)

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx (hunk 7 above)
Comment (MFSepplive) [reply packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:81]: i know it's long but thats needed to show how it works with a table i think.

Comment (nsams) [reply packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx]: I would suggest to:

- create a basic example where you show the usage of FilterBar together with a table (1-2 filters are enough, no More button, no range filter just basic filter)
- show advanced features as their own example, this without full table integration (as you have already done for 2 features)

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+interface IExampleRow {
+    id: number;
+    model: string;
+    brand: string;
+    color: string;
+}
+
+export const randomTableData = Array.from(Array(5).keys()).map((i): IExampleRow => {
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:54]: This should not be exported

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+
+<Meta title="Docs/Components/Table/FilterBar/FilterBar" />
+
+# Filterbar
+
+The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include Final Form Fields.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:7]:
```suggestion
The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include Final Form fields.
```

File: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+
+<Meta title="Docs/Components/Table/FilterBar/FilterBar" />
+
+# Filterbar
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx:5]: Filterbar or FilterBar? Why do you use both?

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx (hunk 7 above)
Comment (johnnyomair) [reply packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx]: Now the stories are pretty neat 👏🏼

File: packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx (hunk 8 above)
Comment (johnnyomair) [reply packages/admin-stories/src/docs/components/Table/FilterBar/stories/Default.stories.tsx:54]: And I'd call it sampleTableData or exampleTableData

**Pull request closed without merging**